src/modules/csm/csm.py

```diff
@@ -255,8 +211,69 @@ def collect_data(self, blockstamp: BlockStamp) -> bool:
 
         return self.state.is_fulfilled
 
+    def calculate_distribution(self, blockstamp: BlockStamp) -> tuple[int, dict[NodeOperatorId, int]]:
+        """Computes distribution of fee shares at the given timestamp"""
+
+        assert self.state
+
+        threshold = self.state.avg_perf - self.w3.csm.oracle.perf_leeway(blockstamp.block_hash)
+        operators_to_validators = self.module_validators_by_node_operators(blockstamp)
+
+        # Build the map of the current distribution operators.
+        distribution: dict[NodeOperatorId, int] = {}
+        for (_, no_id), validators in operators_to_validators.items():
+            if no_id in self.stuck_operators(blockstamp):
```

Comment:
`self.stuck_operators(blockstamp)` should be got outside `for`

Reply:
It's cached, so no need for an extra variable.

Reply:
as a reader it is not clear that there is caching

Reply:
Bad reader.